src/DependencyInjection/SurfnetStepupExtension.php

```diff
@@ -119,19 +121,23 @@ private function configureLocaleCookieSettings(array $config, ContainerBuilder $
      */
     private function configureGatewayApiClient(array $config, ContainerBuilder $container)
     {
+        $handlerStack = new HandlerStack;
```

Comment:
This is not possible as the container cannot deal with instances as parameters before it has been built. Service definitions should be used. Instead of building our own handler stack to prevent exceptions from being thrown on valid HTTP status codes, we should use the default with the [http-errors](http://docs.guzzlephp.org/en/latest/request-options.html#http-errors) options set to false.